src/ws/cockpitcreds.c

```diff
@@ -219,6 +224,43 @@ cockpit_creds_get_csrf_token (CockpitCreds *creds)
 }
 
 /**
+ * cockpit_creds_set_login_data
+ * @creds: the credentials
+ * @object: A JsonObject to store
+ *
+ * It is an error to call this function
+ * if login data has already been set.
+ *
+ */
+
+void
+cockpit_creds_set_login_data (CockpitCreds *creds,
+                              JsonObject *object)
```

Comment:
The idea is that CockpitCreds is an immutable class ... with the exception of the atomic cockpit_creds_poison() stuff, which just sets a flag.